src/osd/PG.cc

```diff
@@ -2789,10 +2797,38 @@ int PG::_prepare_write_info(map<string,bufferlist> *km,
 			    bool dirty_big_info,
 			    bool dirty_epoch)
 {
+  if (dirty_epoch) {
+    ::encode(epoch, (*km)[epoch_key]);
+  }
+
+  // try to do info efficiently?
+  if (!dirty_big_info) {
+    pg_fast_info_t fast;
+    fast.populate_from(info);
+    fast.apply_to(&last_written_info);
+    if (info == last_written_info) {
+      ::encode(fast, (*km)[fastinfo_key]);
+      return 0;
+    }
+    generic_derr << __func__ << " fastinfo failed, info:\n";
+    {
+      JSONFormatter jf(true);
+      jf.dump_object("info", info);
+      jf.flush(*_dout);
+    }
+    {
+      *_dout << "\nlast_written_info:\n";
+      JSONFormatter jf(true);
+      jf.dump_object("last_written_info", last_written_info);
+      jf.flush(*_dout);
+    }
+    *_dout << dendl;
+  }
```

Comment:
@liewegas working fine , no more log flooding , fastinfo size is now 186 bytes..